be/test/vec/aggregate_functions/agg_min_max_by_test.cpp

```diff
@@ -0,0 +1,107 @@
+
+class AggMinMaxByTest : public ::testing::TestWithParam<std::string> {};
+
+TEST_P(AggMinMaxByTest, min_max_by_test) {
+    std::string min_max_by_type = GetParam();
+    // Prepare test data.
+    auto column_vector_value = ColumnInt32::create();
+    auto column_vector_key_int32 = ColumnInt32::create();
+    auto column_vector_key_str = ColumnString::create();
+    auto max_pair = std::make_pair<std::string, int32_t>("foo_0", 0);
+    auto min_pair = max_pair;
+    for (int i = 0; i < agg_test_batch_size; i++) {
+        column_vector_value->insert(cast_to_nearest_field_type(i));
+        column_vector_key_int32->insert(cast_to_nearest_field_type(agg_test_batch_size - i));
+        std::string str_val = fmt::format("foo_{}", i);
+        if (max_pair.first < str_val) {
+            max_pair.first = str_val;
+            max_pair.second = i;
+        }
+        if (min_pair.first > str_val) {
+            min_pair.first = str_val;
+            min_pair.second = i;
+        }
+        column_vector_key_str->insert(cast_to_nearest_field_type(str_val));
+    }
+
+    // Prepare test function and parameters.
+    AggregateFunctionSimpleFactory factory;
+    register_aggregate_function_min_max_by(factory);
+
+    // Test on 2 kind of key types (int32, string).
+    for (int i = 0; i < 2; i++) {
+        DataTypes data_types = {std::make_shared<DataTypeInt32>(),
+                                i == 0 ? (DataTypePtr)std::make_shared<DataTypeInt32>()
+                                       : (DataTypePtr)std::make_shared<DataTypeString>()};
+        Array array;
+        auto agg_function = factory.get(min_max_by_type, data_types, array);
+        std::unique_ptr<char[]> memory(new char[agg_function->size_of_data()]);
+        AggregateDataPtr place = memory.get();
+        agg_function->create(place);
+
+        // Do aggregation.
+        const IColumn* columns[2] = {column_vector_value.get(),
+                                     i == 0 ? (IColumn*)column_vector_key_int32.get()
+                                            : (IColumn*)column_vector_key_str.get()};
+        for (int j = 0; j < agg_test_batch_size; j++) {
+            agg_function->add(place, columns, j, nullptr);
+        }
+
+        // Check result.
+        ColumnInt32 ans;
+        agg_function->insert_result_into(place, ans);
+        if (i == 0) {
+            // Key type is int32.
+            ASSERT_EQ(min_max_by_type == "max_by" ? 0 : agg_test_batch_size - 1,
+                      ans.get_element(0));
+        } else {
+            // Key type is string.
+            ASSERT_EQ(min_max_by_type == "max_by" ? max_pair.second : min_pair.second,
+                      ans.get_element(0));
+        }
+        agg_function->destroy(place);
+    }
+}
+
+INSTANTIATE_TEST_SUITE_P(Params, AggMinMaxByTest,
+                         ::testing::ValuesIn(std::vector<std::string> {"min_by", "max_by"}));
+} // namespace doris::vectorized
+
+int main(int argc, char** argv) {
```

Comment:
We changed unit test running mechanism, not every unit test has a main function.

Reply:
cleared